### ISSUE TITLE
SSH connection errors are unclear

### ISSUE THREAD

**Issue by shawnferry** [ansible/ansible#11041]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.9 (devel ede94dd7f8) last updated 2015/03/29 14:04:17 (GMT -400)
  lib/ansible/modules/core: (detached HEAD de4c7f8ba6) last updated 2015/05/19 12:48:28 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD cb848fcd9e) last updated 2015/04/03 15:14:36 (GMT -400)
  v2/ansible/modules/core: (detached HEAD 34784b7a61) last updated 2015/02/26 12:26:47 (GMT -400)
  v2/ansible/modules/extras: (detached HEAD 46e316a20a) last updated 2015/04/03 15:14:36 (GMT -400)
  configured module search path = None
##### Ansible Configuration:

Nothing of note
##### Environment:

NA
##### Summary:

Hosts without name resolution result in failures with unclear ssh connection errors.
##### Steps To Reproduce:

Attempt to connect to a host in inventory which does not have name resolution. 
##### Expected Results:

Ansible with ssh errors more clearly indicates the reason for failure
###### ansible paramiko

```
fatal: [hostA] => {'msg': 'FAILED: [Errno 8] nodename nor servname provided, or not known', 'failed': True}
```
###### command line ssh

```
ssh hostA
ssh: Could not resolve hostname hostA: nodename nor servname provided, or not known
```
##### Actual Results:

Ansible with ssh errors are unclear
###### ansible ssh

```
fatal: [hostA] => SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh
```
##### Example

If I have hundreds or thousands of hosts some of which need to be bootstrapped in a specific way.  I can't tell from the ssh errors if they need to be bootstrapped or if there is some other error which needs to be addressed separately.  In the simple case this forces me to fall back to paramiko and bootstrap them all again which is slow.

**Comment by bcoca** [comment]
so in 1.9.1 iirc, we added passing more data from the ssh command line to the ssh errors, please try this with the 1.9 stable version or current devel (2.0), you seem to have an older checkout of devel.

**Comment by shawnferry** [comment]
Updated I'm reporting 
     ansible 2.0.0 (devel 5658ddf02e) last updated 2015/05/31 18:48:20 (GMT -400)
       lib/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/05/31 18:53:16 (GMT -400)
       lib/ansible/modules/extras: (detached HEAD 1276420a3a) last updated 2015/05/31 18:53:16 (GMT -400)
       v2/ansible/modules/core: (detached HEAD 85c8a892c8) last updated 2015/05/31 18:53:17 (GMT -400)
       v2/ansible/modules/extras: (detached HEAD 70ea058563) last updated 2015/05/31 18:53:17 (GMT -400)
       configured module search path = None

I'm still seeing 
    SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh
when ssh returns
    ssh: Could not resolve hostname foo: nodename nor servname provided, or not known

Notably now my version is returning 2.0.0 when sourced from hacking/env-setup did I miss something here?

**Comment by bcoca** [comment]
no, my surprise was that it was returning 1.9 from devel

**Comment by shawnferry** [comment]
Finding a slightly different behavior in the most recent commits

{"msg": "ERROR! SSH encountered an unknown error. The output was:\n", "failed": true}

ssh foo
ssh: Could not resolve hostname foo: nodename nor servname provided, or not known

ansible 2.0.0 (devel 1c8527044b) last updated 2015/06/03 14:38:43 (GMT -400)
  lib/ansible/modules/core: (detached HEAD b138411671) last updated 2015/06/03 11:55:57 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 1276420a3a) last updated 2015/06/03 11:56:08 (GMT -400)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/06/03 11:56:18 (GMT -400)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/06/03 11:56:29 (GMT -400)
  configured module search path = None

**Comment by bcoca** [comment]
fixed via bdba807fd1b03d888db6ad19d13cc3f6ec47f968, which restores the previous behavior
